File: Additive Cipher.cpp
#include<iostream>
#include<string.h>
#include<cmath>
using namespace std;

string encrypt(string text, int key) {
	string cipher = "";
	int l = text.length();
	for(int i = 0; i < l; i++) cipher += ((text[i] + key - 65) % 26 + 65);
	return cipher;
}

string decrypt(string cipher, int key) {
	string text = "";
	int l = cipher.length();
	for(int i = 0; i < l; i++) {
		int difference = (cipher[i] - key - 65) % 26;
		if(difference < 0) difference = 26 + difference;
		text += (difference + 65);
	}
	return text;
}

int main() {
	string text = "", encrypted = "", decrypted = "";
	int key;

	// text = "CRYPTO";
	// key = 5;

	cout << "\nEnter plain-text: ";
	cin >> text;
	cout << "Enter key: ";
	cin >> key;

	encrypted = encrypt(text, key);
	cout << "\nEncrypted text: " << encrypted << "\n";
	decrypted = decrypt(encrypted, key);
	cout << "\nDecrypted text: " << decrypted << "\n\n";
}

File: Additive Cipher.py
def encrypt(text, key):
    encrypted = ""
    for character in text:
        encrypted += chr((ord(character) + key - 65) % 26 + 65)
    return encrypted

def decrypt(cipher, key):
    decrypted = ""
    for character in cipher:
        decrypted += chr((ord(character) - key - 65) % 26 + 65)
    return decrypted

# text = "CRYPTO"
# key = 5

text = input("\nEnter the text to be encrypted: ")
key = int(input("Enter key: "))

encrypted = encrypt(text, key)
print("\nEncrypted: " + encrypted)
decrypted = decrypt(encrypted, key)
print("\nDecrypted: " + decrypted + "\n")